prime-angular-frontend/src/app/modules/enrolment/pages/self-declaration/self-declaration.component.html

```diff
@@ -86,6 +86,12 @@
                                  (remove)="onRemove('hasConvictionDocumentGuids', $event)">
             </app-document-upload>
 
+            <app-enrollee-property>
+              <app-document-attachment [documents]=getSelfDeclarationDocuments(SelfDeclarationTypeEnum.HAS_CONVICTION)
+                                       (download)="downloadSelfDeclarationDocument($event)">
+              </app-document-attachment>
+            </app-enrollee-property>
```

Comment:
Part of the comment already exists in our current code but doesn't seem to be an issue.

If I answer yes and upload a document, the document will show in the **overview** page in the right place, and if I go back to Self-Declaration page, the documents show up under the right section as well, however; if I change the answer to that question to **No**, the document and description in the Self-Declaration page will disappear properly, but the document still shows up in the overview page even with an answer of **No** and no description. Also, if I go back to the Self-Declaration page and change the answer back to yes, the documents show up again.

I am not sure if we should persist the uploaded documents if the answer is changed to **No**.

Otherwise all is good.

Reply:
Will try to address in this PR, i.e. we should **not** persist the uploaded documents if the answer is changed to No.

Reply:
Just a reminder too, that if something is outside the scope of the ticket it's ok to take these thoughts and put them in another ticket. Since I think that's a little bigger task that what you have laid out here and we are already sudo rolling this ticket into another sprint. (this ticket is a 0.5)

I say this because how documents are handled/finalized is an area that needs attention, but it is a task that needs some planning and is outside the scope of this ticket.

I think this and @johnnyl6 comment below are about the same topic so this refers to both*

Reply:
Alright, created PRIME-2054 to capture this